aea/protocols/generator.py

```diff
@@ -550,12 +552,38 @@ def _setup(self) -> None:
             pure_custom_type: "Custom" + pure_custom_type
             for pure_custom_type in self._all_custom_types
         }
-        if self.protocol_specification.dialogue_config != {}:
-            self._reply = self.protocol_specification.dialogue_config["reply"]  # type: ignore
+
+        # Dialogue attributes
+        if (
+            self.protocol_specification.dialogue_config != {}
+            and self.protocol_specification.dialogue_config is not None
+        ):
+            self._reply = self.protocol_specification.dialogue_config["reply"]
             roles_set = self.protocol_specification.dialogue_config["roles"]  # type: ignore
             self._roles = sorted(roles_set, reverse=True)
             self._end_states = self.protocol_specification.dialogue_config["end_states"]  # type: ignore
 
+            # infer initial performative
+            set_of_all_performatives = set(self._reply.keys())
+            set_of_all_replies = set()
+            for _, list_of_replies in self._reply.items():
+                set_of_replies = set(list_of_replies)
+                set_of_all_replies.update(set_of_replies)
+            initial_performative_set = set_of_all_performatives.difference(
+                set_of_all_replies
+            )
+            initial_performative_list = list(initial_performative_set)
+            if len(initial_performative_list) != 1:
+                raise ProtocolSpecificationParseError(
+                    "Invalid reply structure. There must be a single speech-act which is not a valid reply to any other speech-acts so it can be designated as the inital speech-act. Found {} of such speech-acts in the specification".format(
+                        len(initial_performative_list),
+                    )
+                )
```

Comment:
interesting and makes sense given current setup, can we think of a dialogue where this restriction might be a problem?

Reply:
Well, on second thought, the above is kinda restrictive (works with fipa case though). Imaging a simple generic dialogue where the two agents just send `inform(msg)` to each other then: 

```
speech-acts:
  inform: msg: ct:Message
```

and 

```
reply:
inform: {inform}
```

In this case, the above condition fails since `len(initial_performative_list) == 0`.

Reply:
In the first case, `initial_performative_list=[inform]`..although I agree, your current implementation would raise an error.

In general, we should probably allow for more than one initial performative. So this could make the list of potential additions in future.